### ISSUE TITLE
Bad error mesasage: "unsupported features were requested: (empty)"

### ISSUE THREAD

**Issue by Craig-Macomber** [gfx-rs/wgpu#1681]
**Description**
wgpu-rs reported the error:

> 'Error in Adapter::request_device: unsupported features were requested: (empty)', /home/craig/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.9.0/src/backend/direct.rs:113:9

**Repro steps**

Create a device with `features: wgpu::Features::ALL_NATIVE`. Then remove any features reported as not supported (for example in my case that was:

```rust
features: wgpu::Features::ALL_NATIVE
                    ^ wgpu::Features::TEXTURE_COMPRESSION_ASTC_LDR
                    ^ wgpu::Features::VERTEX_ATTRIBUTE_64BIT,
```

**Expected vs observed behavior**

Expected: here are unsupported features, and the error lists them, or there aren't and the error should not occur.

Observed: error says there are unsupported features, but lists them as `(empty)`.

**Extra materials**
I suspect this error is due to do unused feature bits being set. Either these should be ignored, or reported better in the error message.

**Platform**
Linux running wgpu 0.9.0 natively on vulkan.

**Comment by Craig-Macomber** [comment]
I took a look at this on master to see if I could make an easy fix. Looks like its already non an issue there.

`wgpu::Features::ALL_NATIVE` has been moved to a function with a better name (`wgpu::Features::all_native_mask()`).

Additionally this case (where you opt into non-existant features via the mask) no longer errors.

Thus I consider this fixed on master. I'm not sure if you want this bug closed now, or once its released, so I'll leave it open for now.
